Sources/AWSLambdaDeploymentDescriptorGenerator/JSONSChemaReader.swift

```
@@ -44,116 +84,329 @@ enum JSONPrimitiveType: Decodable, Equatable {
-        }
-        switch onlyKey {
-        case .type:
-            let primitiveType = try container.decode(JSONPrimitiveType.self, forKey: .type)
-            self = .type(primitiveType)
-        case .ref:
-            let ref = try container.decode(String.self, forKey: .ref)
-            self = .ref(ref)
+        
+        if let type = try? container.decode([JSONPrimitiveType].self, forKey: .type) {
+            self = .array(type)
+        } else {
+            if let singleType = try? container.decode(JSONPrimitiveType.self, forKey: .type) {
+                self = .single(singleType)
+            } else {
+                throw DecodingError.typeMismatch(JSONSchema.self, DecodingError.Context(codingPath: container.codingPath, debugDescription: "➡️ Could not decode 'type' property. Expected either an array of primitive types or a single primitive type", underlyingError: nil))
+                
+            }
         }
     }
 }
 
+
+//MARK: - JSONUnionType ..
+
 enum JSONUnionType: Decodable {
     case anyOf([JSONType])
     case allOf([JSONType])
     case type(JSONType)
+    case reference(String)
+    
     
     enum CodingKeys: String, CodingKey {
-        case anyOf
-        case allOf
+        case anyOf = "anyOf"
+        case allOf = "allOf"
+        case ref = "$ref"
     }
     
-    init(from decoder: any Decoder) throws {
+    
+    init(from decoder: Decoder) throws {
         let container = try decoder.container(keyedBy: CodingKeys.self)
-        var allKeys = ArraySlice(container.allKeys)
-        if let onlyKey = allKeys.popFirst(), allKeys.isEmpty  {
-            // there is an anyOf or allOf key
-            switch onlyKey {
-            case .allOf: fatalError("not yet implemented")
-            case .anyOf:
-                let value = try container.decode(Array<JSONType>.self, forKey: .anyOf)
-                self = .anyOf(value)
+        do {
```

not necessary to catch the error here, just use `try` and let errors propagate up to the stack